spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindReturnRef.java

```diff
@@ -182,8 +194,8 @@ public void sawOpcode(int seen) {
             OpcodeStack.Item item = stack.getStackItem(0);
             XField field = item.getXField();
             if (method == null || !"duplicate".equals(method.getName())
-                    || !method.getSignature().matches("\\(\\)Ljava/nio/[A-Za-z]+Buffer;")
-                    || !method.getClassDescriptor().getSignature().matches("Ljava/nio/[A-Za-z]+Buffer;")) {
+                    || !duplicateMethodSignatureMatcher.reset(method.getSignature()).matches()
```

Comment:
This is a curious approach. Normally I would just create a constant `Pattern` and then generate a new `Matcher` on each use. Is this more efficient?

Reply:
Yes, that was also my intention. But then I met this: https://www.baeldung.com/java-regex-pre-compile. In this study there is a benchmark which claims that creating a matcher instead and resetting it on each use is almost 2 time faster. I think it is no wonder, since creation of a whole new Matcher class is definitely more expensive than just replacing its string reference to another string. However, if you think that this is too hard to understand for other developers then I can change it to pattern-only.